### PR TITLE
PS-5686: improve innodb_redo_log_encrypt validation and formatting (8.0).

### PR DESCRIPTION
**Pull request by dutow**
* Variable no longer accepts 2 and 3 as valid values
* Value is dispayed in upcase as with other variables

(cherry picked from commit 5524aa3c84ced75558a10379fd90c7b74cc1be63)

### CODE REVIEW

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
Same comments as for the 5.7 version

File: storage/innobase/handler/ha_innodb.cc [hunk 1]
```diff
@@ -24396,6 +24408,61 @@ static int innodb_encrypt_tables_validate(
...
+@param[in]	value	incoming string
+@return 0 on success, 1 on failure.
+*/
+static int innodb_redo_log_encrypt_validate(THD *thd, SYS_VAR *var, void *save,
+                                            struct st_mysql_value *value) {
+  const char *redo_log_encrypt_input;
+  char buff[STRING_BUFFER_USUAL_SIZE];
+  int len = sizeof(buff);
+
+  ut_a(save != NULL);
```
Comment (laurynas-biveinis) [inline storage/innobase/handler/ha_innodb.cc]:
```suggestion
  ut_a(save != nullptr);
```

File: storage/innobase/handler/ha_innodb.cc [hunk 2]
```diff
@@ -24396,6 +24408,61 @@ static int innodb_encrypt_tables_validate(
...
+  ut_a(value != NULL);
```
Comment (laurynas-biveinis) [inline storage/innobase/handler/ha_innodb.cc]:
```suggestion
  ut_a(value != nullptr);
```

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
OK with the minor change, no need for further reviews

**Comment by dutow** [comment]
Rebased on 8.0 after ps-5565 merge, no other change. 

https://ps80.cd.percona.com/job/percona-server-8.0-param/216/

**Comment by dutow** [comment]
Rerecorded log_encrypt_2 and reran jenkins: https://ps80.cd.percona.com/job/percona-server-8.0-param/224

Failures seems to be jenkins timeouts around 99% mtr, previous run completed all.

**Pull request merged**